tests/cross_functional/system_test/test_object_expiration.py

```diff
@@ -1,96 +1,426 @@
-from ocs_ci.framework.testlib import skipif_ocs_version
-from ocs_ci.ocs.bucket_utils import s3_put_object, s3_get_object
+from ocs_ci.framework.testlib import version, skipif_ocs_version
+from ocs_ci.ocs.bucket_utils import (
+    s3_put_object,
+    upload_bulk_buckets,
+    expire_objects_in_bucket,
+    s3_list_objects_v2,
+    bulk_s3_put_bucket_lifecycle_config,
+)
+from ocs_ci.ocs.resources.pod import (
+    get_noobaa_core_pod,
+    get_noobaa_db_pod,
+    wait_for_noobaa_pods,
+)
+from ocs_ci.ocs.node import (
+    drain_nodes,
+    wait_for_nodes_status,
+    schedule_nodes,
+    get_node_objs,
+)
+from ocs_ci.utility.utils import TimeoutSampler
 
 logger = logging.getLogger(__name__)
 
 
-@magenta_squad
-@system_test
-@bugzilla("2039309")
-@skipif_ocs_version("<4.11")
-@pytest.mark.polarion_id("OCS-4852")
-def test_object_expiration(mcg_obj, bucket_factory):
-    """
-    Test object expiration, see if the object is deleted within the expiration + 8 hours buffer time
-
-    """
-    # Creating S3 bucket
-    bucket = bucket_factory()[0].name
-    object_key = "ObjKey-" + str(uuid.uuid4().hex)
-    obj_data = "Random data" + str(uuid.uuid4().hex)
-    expiration_days = 1
-    buffer_time_in_hours = 8
-
-    expire_rule_4_10 = {
-        "Rules": [
-            {
-                "Expiration": {
-                    "Days": expiration_days,
-                    "ExpiredObjectDeleteMarker": False,
-                },
-                "ID": "data-expire",
-                "Prefix": "",
-                "Status": "Enabled",
-            }
-        ]
-    }
-    expire_rule = {
-        "Rules": [
-            {
-                "Expiration": {
-                    "Days": expiration_days,
-                    "ExpiredObjectDeleteMarker": False,
-                },
-                "Filter": {"Prefix": ""},
-                "ID": "data-expire",
-                "Status": "Enabled",
-            }
-        ]
-    }
+class TestObjectExpiration:
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
 
-    logger.info(f"Setting object expiration on bucket: {bucket}")
-    if version.get_semantic_ocs_version_from_config() < version.VERSION_4_11:
-        mcg_obj.s3_client.put_bucket_lifecycle_configuration(
-            Bucket=bucket, LifecycleConfiguration=expire_rule_4_10
+    def check_if_objects_expired(self, mcg_obj, bucket_name, prefix=""):
+        response = s3_list_objects_v2(
+            mcg_obj, bucketname=bucket_name, prefix=prefix, delimiter="/"
         )
-    else:
-        mcg_obj.s3_client.put_bucket_lifecycle_configuration(
-            Bucket=bucket, LifecycleConfiguration=expire_rule
+        return response["KeyCount"] == 0
+
+    @system_test
+    @bugzilla("2039309")
+    @skipif_ocs_version("<4.11")
+    @pytest.mark.polarion_id("OCS-4852")
+    @magenta_squad
+    def test_object_expiration(
```

Comment:
this appears to be a functional test? any reason for placing it under system_test

Reply:
yeah, back then we decided to keep it under system test because of the execution time